src/front/app/features/challenge/components/Challenge.js

```diff
@@ -57,7 +70,7 @@ class Challenge extends Component {
                     menu={<ChallengeDetailsMenu
                         {...Object.assign(this.state.menu, {
                             onWatchChallenge: () => onWatchChallenge(challenge.get('_id')),
-                            isWatched: challenge.get('isWatched')
+                            isWatched: false
```

Comment:
Maybe it is better to use `true` as a default? In this case "Add to watch list" icon appears only when it is needed rather than hides after watchlist is loaded.